clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/common/model/AzureDeploymentOperation.groovy

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2015 The original authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.azure.resources.common.model
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.microsoft.azure.management.resources.models.DeploymentOperation
+import com.netflix.spinnaker.clouddriver.azure.client.AzureResourceManagerClient
+import com.netflix.spinnaker.clouddriver.azure.security.AzureCredentials
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import groovy.util.logging.Slf4j
+
+@Slf4j
+class AzureDeploymentOperation {
+  static ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+
+  static AzureDeploymentOperation getObjectFromJson(String responseContent) {
+    mapper.readValue(responseContent, AzureDeploymentOperation.class)
+  }
+
+  /**
+   * Return a collection of deployment related errors (empty list if deployment was successful)
+   * @param task the current task to be updated
+   * @param creds the credentials to use when communicating to the Azure subscription(s)
+   * @param resourceGroupName the resource group where we try to deploy
+   * @param deploymentName the name of the deployment
+   * @return a Collection of error messages capturing deployment related errors
+   */
+  static List<String> checkDeploymentOperationStatus(Task task, String opsName, AzureCredentials creds, String resourceGroupName, String deploymentName) {
+    def errList = new ArrayList<String>()
+    Map<String, Boolean> resourceCompletedState = new HashMap<String, Boolean>()
+    String deploymentState = AzureResourceManagerClient.DeploymentState.DEPLOYING
+    Boolean checkDeployment = true
+
+    while (checkDeployment) {
```

Comment:
I'd like to see an upper bounds on how long you're going to loop for ... rather than the infinite wait